maps/MODULES/locations_1.lua

```diff
@@ -0,0 +1,33 @@
+local locations = {}
+
+function locations.CyborgUpgradeLocation()
+	h = GetCurrentHero()
+	goblins = tonumber(GetCreatureAmount(h, CREATURE_GOBLIN)) or 0
+	-- someone else
+	pikemen = tonumber(GetCreatureAmount(h, CREATURE_PIKEMAN)) or 0
+	if (goblins + pikemen) == 0 then
+		AdvancedMessageBox("{Hill Fort}\n\nAn unusual alliance of Cyborgs to train (upgrade) Pikemen and Goblins.  Unfortunately, you have none with you.",
+	  1, -1, -1, -1, -1, -1, -1, -1, 0)
+	else
+		if goblins > 0 then 
+			TakeTroop(h, CREATURE_GOBLIN, goblins)
+			GrantArmy(h, CREATURE_CYBER_KOBOLD_SPEARMAN, goblins)
+		end
+		-- someone else
+		if pikemen > 0 then
+			TakeTroop(h, CREATURE_PIKEMAN, pikemen)
+			GrantArmy(h, CREATURE_CYBER_PLASMA_LANCER, pikemen)
+		end
+		PlaySoundEffect('EXPERNCE')
+		if goblins > 0 and pikemen > 0 then
+			AdvancedMessageBox("{Hill Fort}\n\nAll of your Pikemen and Goblins have been upgraded to Cyborg Plasma Lancers and Cyborg Kobold Spearmen", 1, -1, -1, -1, -1, -1, -1, -1, 0)
```

Comment:
Can you show an image of these lancers and spearmen?

https://github.com/jkoppel/project-ironfist/blob/master/maps/MODULES/AdvMessageBox.lua#L12
http://projectironfist.pbworks.com/w/page/134600181/AdvancedMessageBox%20Documentation

Similar for the other branches.